### ISSUE TITLE
Align GLSL matrix multiply and divide with OSL and C++

### ISSUE THREAD

**Issue by HardCoreCodin** [AcademySoftwareFoundation/MaterialX#2089]
Shading languages use different matrix conventions: OSL is row major, GLSL is column major
This determines multiplication order: `M1 * M2` in OSL is `M2 * M1` in GLSL
MaterialX back-end implementations generate code with the same ordering, consistent with the node inputs.
If it is correct for OSL, it would be incorrect for GLSL (and vice versa)
This makes the same node (and hence any graph that uses it) non-portable across back-ends.

<img width="973" alt="MatMul_Blender_Vs_MGE_REVERSED" src="https://github.com/user-attachments/assets/64a4f1e7-212d-4b3f-b177-b779db72c0cf">
<img width="983" alt="MatMul_Blender_Vs_MGE" src="https://github.com/user-attachments/assets/72553ebf-43f5-4f06-aaf4-a1b3a63b20b8">
<img width="906" alt="MatMul_Blender_Vs_MGE_M1" src="https://github.com/user-attachments/assets/90dd43bc-1ee7-412f-ae41-d105f648eaa9">
<img width="979" alt="MatMul_Blender_Vs_MGE_M2" src="https://github.com/user-attachments/assets/e7657022-18db-4e3b-bb8b-365d9444b86d">

MaterialX specification mentions row major, but only with reference to matrix literal construction component ordering.
However, both OSL and GLSL use the same ordering anyway: ( X.x, X.y, X.z , Y.x, Y.y, Y.z ,  Z.x, Z.y, Z.z )
So this is not really saying anything meaningful.
The specification should state what the implied matrix convention is in MaterialX - explicitly.
All implementations should then be made to account for that convention.

If it is row major, then the ordering for the `multiply` node types that take a 3x3 or 4x4 matrix, should be reversed in GLSL.

<img width="917" alt="MatrialX_glsl_impl_incorrect_multiply_divide_matrix" src="https://github.com/user-attachments/assets/691ba1b2-45dd-480f-8b93-b37bfdd44793">

There should also be a special handling for `divide` node types, as just flipping the order will not suffice:
Matrix division is defined as the first matrix provided multiplied by the inverse of the second.
The correct matrix need to be inverted in column-major implementations.

<img width="931" alt="MGE_matrix_A_divided_by_B_flipped_order" src="https://github.com/user-attachments/assets/256d7ad9-9e60-40c6-8e5e-35ea57e487ac">
<img width="873" alt="MGE_matrix_A_multiply_by_inverse_of_B_flipped_order" src="https://github.com/user-attachments/assets/2ef1afb4-ab14-4af7-84d4-f744a3cdc867">

**Comment by dbsmythe** [comment]
The MaterialX Specification only states that the constructor values for a matrix are "9 or 16 floats separated by commas, specified in row-major order".  It doesn't (and I believe **shouldn't**) state whether the resulting matrix is row-major or column-major ordering in memory; instead, each shading language should use whatever its native representation for a matrix is.  So anything that constructs a matrix (either direct values or the &lt;creatematrix> node) should put the 9 or 16 values in the correct places for an implementation, e.g. the 13th, 14th and 15th values in a matrix contructor tuple should go in the 1st/2nd/3rd spots in the bottom row of the matrix, however that's actually laid out in memory.  Similarly for anything that reads or extracts specific values from within a matrix: we should provide nodes to do that in a language-independent way (e.g. there is a proposed &lt;extractrowvector> node, there should also be an &lt;extractcolumnvector> node).

**Comment by jstone-lucasfilm** [comment]
Good catch, @HardCoreCodin, and we've proposed to address this by updating the GLSL implementations of matrix multiply and divide to align with the OSL behavior.

The OSL behavior of the matrix product additionally aligns with our own reference implementation in MaterialX C++, which is defined here:

https://github.com/AcademySoftwareFoundation/MaterialX/blob/main/source/MaterialXCore/Types.h#L511

**Comment by jstone-lucasfilm** [comment]
Thanks for this report, @HardCoreCodin, as well as to @tdavidovicNV for the fix in #2298.
